pkgs/applications/networking/instant-messengers/cinny/default.nix

```diff
@@ -29,6 +29,9 @@ buildNpmPackage rec {
 
   npmDepsHash = "sha256-4R+To2LhcnEM9x1noo6MhCckyBKgPWiAi7zgDqAmaN0=";
 
+  # Fix error: no member named 'aligned_alloc' in the global namespace
+  env.NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.isDarwin && stdenv.isx86_64) "-D_LIBCPP_HAS_NO_LIBRARY_ALIGNED_ALLOCATION=1";
```

Comment:
so this is due to darwin x64 defaulting to the 10.12 sdk and aligned_alloc only available in later versions, 10.15 maybe? aarch64-darwin uses the 11 sdk, which is why this fails on x64 and not aarch64.

using `overrideSDK` one can change the sdk version used though it seems somewhat tricky to get correct, at least that's been my experience.

anyway, i got the build to work using the diff bellow. though, i'm not clear if that is correct usage as overrideSDK can take a set, eg: ` { darwinMinVersion = "10.7";  darwinSdkVersion = "11.0"; }` to also set the minVersion, where just using a string only sets the SdkVersion.


<details><summary>overridesdk.diff</summary>

```diff
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 8f4e2cba31d5..e17da4365549 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -4566,7 +4566,15 @@ with pkgs;
 
   cht-sh = callPackage ../tools/misc/cht.sh { };
 
-  cinny = callPackage ../applications/networking/instant-messengers/cinny { };
+  cinny = let
+    stdenv' = if stdenv.isDarwin && lib.versionOlder stdenv.hostPlatform.darwinSdkVersion "11"
+        then overrideSDK stdenv "11.0"
+        else stdenv;
+  in
+  callPackage ../applications/networking/instant-messengers/cinny {
+    buildNpmPackage = buildNpmPackage.override { stdenv = stdenv'; };
+    stdenv = stdenv';
+  };
 
   cinny-desktop = callPackage ../applications/networking/instant-messengers/cinny-desktop { };
 
```
</details>

Reply:
forgot to mention. that if you do choose `-D_LIBCPP_HAS_NO_LIBRARY_ALIGNED_ALLOCATION=1` the condition should be something like `stdenv.isDarwin && lib.versionOlder stdenv.hostPlatform.darwinSdkVersion "11.0"` or something.